### PR TITLE
bloodhound-py: init at 1.6.1

### PR DESCRIPTION
**Pull request by exploitoverload**
###### Description of changes
Adds BloodHound.py (also called bloodhound-python) - is an ingestor for Bloodhound written in python that runs on the attacker machine.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Related to #81418

### CODE REVIEW

**Comment by fabaff** [comment]
Successor of https://github.com/NixOS/nixpkgs/pull/241748

File: pkgs/tools/security/bloodhound-py/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  ];
+
+  doCheck = false;
+
+  meta = with lib; {
+      description = "Ingestor for BloodHound";
+      homepage = "https://github.com/fox-it/BloodHound.py";
+      license = licenses.mit;
+      maintainers = with maintainers; [ exploitoverload ];
+    };
```
Comment (figsoda) [inline pkgs/tools/security/bloodhound-py/default.nix:30]:
```suggestion
  meta = with lib; {
    description = "Ingestor for BloodHound";
    homepage = "https://github.com/fox-it/BloodHound.py";
    license = licenses.mit;
    maintainers = with maintainers; [ exploitoverload ];
  };
```

File: pkgs/tools/security/bloodhound-py/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    hash = "sha256-K74VntcRcIgKu48JhLuWeFcqPQg7vurd0+H1e6Kk3Rg=";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [
+    impacket
+    ldap3
+    dnspython
+  ];
+
+  doCheck = false;
```
Comment (figsoda) [inline pkgs/tools/security/bloodhound-py/default.nix:23]: there should be a comment explaining why

Comment (exploitoverload) [reply pkgs/tools/security/bloodhound-py/default.nix]: I think other distributions like blackarch does not execute the tests of bloodhound.py.

Comment (natsukium) [reply pkgs/tools/security/bloodhound-py/default.nix]: I think this package does not provide tests in the first place.

Comment (exploitoverload) [reply pkgs/tools/security/bloodhound-py/default.nix]: Right, I just checked the repository and it does not contain any tests.

File: pkgs/tools/security/bloodhound-py/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    hash = "sha256-K74VntcRcIgKu48JhLuWeFcqPQg7vurd0+H1e6Kk3Rg=";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [
+    impacket
+    ldap3
+    dnspython
+  ];
+
+  doCheck = false; # checks fail because of a bad_format_string error.
```
Comment (figsoda) [inline pkgs/tools/security/bloodhound-py/default.nix:23]:
```suggestion
  # the package has no tests
  doCheck = false;
```

File: pkgs/tools/security/bloodhound-py/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{ lib
+, fetchFromGitHub
+, python3
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "bloodhound-py";
+  version = "1.0.1";
```
Comment (natsukium) [inline pkgs/tools/security/bloodhound-py/default.nix:8]: The author hasn't tagged it on the VCS, but they have released 1.6.1.
1.0.1 is an older version, over three years ago.

**Review by natsukium — APPROVED** [review]
LGTM, thanks!

**Review by figsoda — APPROVED** [review]

**Pull request merged**